### PR TITLE
feat: secrets.example.py

### PR DESCRIPTION
**Pull request by wesleyboar**
## Overview

Create a `secrets.example.py`.

## Related

- similar to https://github.com/TACC/Core-CMS/pull/577
- similar to [Core-Portal's `settings_secret.example.py`](https://github.com/TACC/Core-Portal/blob/v3.7.0/server/portal/settings/settings_secret.example.py)

## Changes

- **added** `secrets.example.py`
- **documented** creation of `secrets.py`
- **documented** creation of `settings_custom.py`

## Testing & UI

N/A

### CODE REVIEW

**Review by taoteg — CHANGES_REQUESTED** [review]
A number of these settings are placed in the `*.settings_local.py` file rather than `secrets.py`. 

Specifically:
- `ALLOWED_HOSTS`
- `CEP_AUTH_VERIFICATION_ENDPOINT`

Reason: These settings do not contain sensitive credentials.

Current Example: https://github.com/TACC/Core-Portal-Deployments/blob/main/ami/camino/prod.cms.settings_local.py

Can we split these into the appropriate files (using this same setup but including the `settings_local.py` file as well) so the usage matches the deployment environment?

**Comment by wesleyboar** [comment]
Thanks, yes! Will do. I am glad to have your eyes on this.

File: taccsite_cms/secrets.example.py
Comment (wesleyboar) [inline taccsite_cms/secrets.example.py:31]: @taoteg, should **all** of these stay in `secrets.py`?

- The `ES_AUTH` value seems secret.
- The `ES_…` values are expected by `HAYSTACK_CONNECTIONS`.

**Review by wesleyboar — COMMENTED** [review]
@taoteg I moved secret values to `settings_local.py`. Also, please see my question about `ES_` values.

File: taccsite_cms/secrets.example.py
Comment (taoteg) [reply taccsite_cms/secrets.example.py:31]: Yes, keep in secrets. While only the `ES_AUTH` value is truly sensitive, it is easier to keep them all together for tracking purposes (and avoid leaking any seemingly harmless info about internal portal configs). 

We might should have another review as to "what values live in which files" again, it has been a while AFAIK since we evaluated the specific location of values distributed across the defaults/custom/local/secrets files.

Comment (wesleyboar) [reply taccsite_cms/secrets.example.py:31]: Moving `ES_…` values would break ES if `HAYSTACK…` is in any of those files.

I've hit loading order complexity before. I solved it via [new section in `settings.py` **after** settings/secrets import](https://github.com/TACC/Core-CMS/blob/v4.15.2/taccsite_cms/settings.py#L808-L814).

I think the complexity of `ES_` in many files is "not worth it". I feel safe to do so, but I defer to you.[^1]

[^1]: Would you feel compelled to update such files on servers or in Core-Portal-Deployments?

Comment (taoteg) [reply taccsite_cms/secrets.example.py:31]: I 100% agree that it is "not worth it" due to the second-order effects.

> Would you feel compelled to update such files on servers or in Core-Portal-Deployments?

I don't think any `ES_` values live anywhere _except_ in the `secrets.py` files, so you have to update the file on the individual host (and add any changes to the Stache entry, naturally).

Comment (wesleyboar) [reply taccsite_cms/secrets.example.py:31]: K. So, good to go this PR?

**Review by taoteg — APPROVED** [review]
LGTM!

File: taccsite_cms/secrets.example.py
Comment (taoteg) [reply taccsite_cms/secrets.example.py:31]: my bad, yes - approved!

**Pull request merged**
